src/Dutch.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+pragma solidity ^0.8.30;
+
+import {UD60x18, uEXP2_MAX_INPUT, uUNIT} from "@prb-math/UD60x18.sol";
+
+/**
+ * @dev Recurring dutch auction
+ */
+library Dutch {
+    uint256 public constant RESET_FACTOR = 4;
+    uint256 public constant HALVING_INTERVAL = 3.5 days;
+    uint256 public constant MAX_DECAY = uEXP2_MAX_INPUT * HALVING_INTERVAL / uUNIT;
+
+    /**
+     * @notice Exponential decay by 1/4 per week
+     * @param startPrice The initial price at elapsed = 0
+     * @param elapsed The amount of time since the startPrice
+     */
+    function decay(uint256 startPrice, uint256 elapsed) internal pure returns (uint256 price) {
+        if (elapsed > MAX_DECAY) {
+            return 0;
+        }
+        UD60x18 coefficient = UD60x18.wrap(startPrice);
+        UD60x18 decayFactor = UD60x18.wrap(elapsed * uUNIT / HALVING_INTERVAL).exp2();
+
+        return coefficient.div(decayFactor).unwrap();
+    }
+}
```

Comment:
Seems correct to me, comparing with the formula in https://www.notion.so/filecoindev/Liquidation-Price-Modeling-26cdc41950c180dca05de4c991d56180?source=copy_link#270dc41950c18024b568ebc9341b2d6d 

Basically, we put RESET_FACTOR = 4, halving interval 3.5 days which means startPrice is divided by 4 after 2 halving interval (i.e. 1 week), which matches the formula in the notion with M = 4. (as here we have price = startPrice / 2^(elapsed / HALVING_INTERVAL) ) cc @wjmelements